console-subscriber/examples/grpc_web.rs

```diff
@@ -0,0 +1,191 @@
+//! Run with:
+//! ```sh
+//! cargo run --example grpc_web --features grpc-web
+//! ```
+use std::time::Duration;
+
+use console_subscriber::ConsoleLayer;
+use http::header::HeaderName;
+use tower_http::cors::{AllowOrigin, CorsLayer};
+
+static HELP: &str = r#"
+Example console-instrumented app with gRPC-Web support.
+
+USAGE:
+    app [OPTIONS]
+
+OPTIONS:
+    -h, help    prints this message
+    blocks      Includes a (misbehaving) blocking task
+    burn        Includes a (misbehaving) task that spins CPU with self-wakes
+    coma        Includes a (misbehaving) task that forgets to register a waker
+    noyield     Includes a (misbehaving) task that spawns tasks that never yield
+"#;
+
+const DEFAULT_MAX_AGE: Duration = Duration::from_secs(24 * 60 * 60);
+const DEFAULT_EXPOSED_HEADERS: [&str; 3] =
+    ["grpc-status", "grpc-message", "grpc-status-details-bin"];
+const DEFAULT_ALLOW_HEADERS: [&str; 4] =
+    ["x-grpc-web", "content-type", "x-user-agent", "grpc-timeout"];
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    let cors = CorsLayer::new()
+        .allow_origin(AllowOrigin::mirror_request())
+        .allow_credentials(true)
+        .max_age(DEFAULT_MAX_AGE)
+        .expose_headers(
+            DEFAULT_EXPOSED_HEADERS
+                .iter()
+                .cloned()
+                .map(HeaderName::from_static)
+                .collect::<Vec<HeaderName>>(),
+        )
+        .allow_headers(
+            DEFAULT_ALLOW_HEADERS
+                .iter()
+                .cloned()
+                .map(HeaderName::from_static)
+                .collect::<Vec<HeaderName>>(),
+        );
+    ConsoleLayer::builder()
+        .with_default_env()
+        .with_cors(cors)
+        .init();
+    // spawn optional extras from CLI args
+    // skip first which is command name
+    for opt in std::env::args().skip(1) {
+        match &*opt {
+            "blocks" => {
+                tokio::task::Builder::new()
+                    .name("blocks")
+                    .spawn(double_sleepy(1, 10))
+                    .unwrap();
+            }
+            "coma" => {
+                tokio::task::Builder::new()
+                    .name("coma")
+                    .spawn(std::future::pending::<()>())
+                    .unwrap();
+            }
+            "burn" => {
+                tokio::task::Builder::new()
+                    .name("burn")
+                    .spawn(burn(1, 10))
+                    .unwrap();
+            }
+            "noyield" => {
+                tokio::task::Builder::new()
+                    .name("noyield")
+                    .spawn(no_yield(20))
+                    .unwrap();
+            }
+            "blocking" => {
+                tokio::task::Builder::new()
+                    .name("spawns_blocking")
+                    .spawn(spawn_blocking(5))
+                    .unwrap();
+            }
```

Comment:
What do you think about making this example simpler? This will avoid duplicating so much code from the `app` example.

Alternatively, I would instead extend the `app` example to support gRPC-web based on command line flags and make it the "catch-all-example" that includes everything.

Reply:
I simplified the example only to keep task1 and task2. Also, include a full example for grpc-web with cors. Do you think we need to add another grpc-web with the default config?

Reply:
I don't think we need another full example, but we should have an example in the docs for `serve_with_grpc_web` as I mentioned there.